doc/developers/contributing.rst

```diff
@@ -751,8 +751,11 @@ In general have the following in mind:
 * When editing reStructuredText (``.rst``) files, try to keep line length under
   80 characters when possible (exceptions include links and tables).
 
-* Before submitting you pull request check if your modifications have introduced
-  new sphinx warnings and try to fix them.
+* Do not modify label for internal links as this would break existing cross
+  references in the documentation.
+
+* Before submitting your pull request check if your modifications have
+  introduced new sphinx warnings and try to fix them.
```

Comment:
As manually detecting that warnings are newly introduced can be challenging for first time contributors, maby we can just warn them that our CI will check for this with something like this:

The automated scikit-learn documentation build running on Circle CI will automatically check if your pull request (PR) causes sphinx to issue warnings related to the files you changed in your PR. If that the case, you will need to push a new commit to your PR with the necessary fix for those warnings.

Reply:
This paragraph has been added in #14967. And those are very synthetic items: if you think that this is too difficult for new contributors I'd rather remove it. Perhaps more details about the CircleCI check should go [there](https://scikit-learn.org/stable/developers/contributing.html#continuous-integration-ci)?

Reply:
As you wish we can leave it here as it is. I was just suggesting an improvement to try to be friendlier to the user who will read the doc building doc for the first time but will not necessarily the section on CI.